src/plots/linePlot.ts

```diff
@@ -61,6 +140,136 @@ export module Plots {
       return new Plottable.Drawers.Line(dataset);
     }
 
+    protected _extentsForProperty(property: string) {
+      let extents = super._extentsForProperty(property);
+
+      if (!this._autorangeSmooth) {
+        return extents;
+      }
+
+      if (this.autorangeMode() !== property) {
+        return extents;
+      }
+
+      if (this.autorangeMode() !== "x" && this.autorangeMode() !== "y") {
+        return extents;
+      }
+
+      let edgeIntersectionPoints = this._getEdgeIntersectionPoints();
+      let includedValues: number[];
+      if (this.autorangeMode() === "y") {
+        includedValues = edgeIntersectionPoints.left.concat(edgeIntersectionPoints.right).map((point) => point.y);
+      } else { // === "x"
+        includedValues = edgeIntersectionPoints.top.concat(edgeIntersectionPoints.bottom).map((point) => point.x);
+      }
+
+      let maxIncludedValue = Math.max.apply(this, includedValues);
+      let minIncludedValue = Math.min.apply(this, includedValues);
+
+      extents = extents.map((extent) => {
+        // When no end-points are in the viewport
+        if (extent.length === 0) {
+          return [minIncludedValue, maxIncludedValue];
+        }
+        return [Math.min(minIncludedValue, extent[0]), Math.max(maxIncludedValue, extent[1])];
+      });
+
+      return extents;
+    }
+
+    private _getEdgeIntersectionPoints(): EdgeIntersection {
+      if (!(this.y().scale instanceof QuantitativeScale && this.x().scale instanceof QuantitativeScale)) {
+        return {
+          top: [],
+          bottom: [],
+          left: [],
+          right: []
+        };
+      }
+
+      let yScale = <QuantitativeScale<number>>this.y().scale;
+      let xScale = <QuantitativeScale<any>>this.x().scale;
+
+      let intersectionPoints: EdgeIntersection = {
+        top: [],
+        bottom: [],
+        left: [],
+        right: []
+      };
+      let leftX = xScale.scale(xScale.domain()[0]);
+      let rightX = xScale.scale(xScale.domain()[1]);
+      let bottomY = yScale.scale(yScale.domain()[0]);
+      let topY = yScale.scale(yScale.domain()[1]);
+
+      this.datasets().forEach((dataset) => {
```

Comment:
This looks like an `O(total number of data points)` algorithm, because we don't have an assurance that the points are sorted in either X or Y. I'm beginning to think that `Line` should have been built like `Bar`, with an explicit orientation.

What's the most common use-case here? Can we add the smoothing in other directions later? Right now it looks like  the most common use case (left-to-right sorted) has its runtime reduced from `log(n)` to `(n)` because of the need to make it universal. We have precedent for considering this -- `entityNearest()` uses X-then-Y logic, for example.